### PR TITLE
Start switching to Github Actions for CI

### PR DESCRIPTION
**Pull request by JonRowe**

### CODE REVIEW

File: ci/.github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: RSpec CI
+on: [pull_request, push]
```
Comment (pirj) [inline ci/.github/workflows/ci.yml:2]: I believe this needs to be tweaked so we only run `push` on `main`/`*-maintenance`/`*-dev`

**Review by pirj — APPROVED** [review]
Awesome!
We can deal with double build problem later on.

File: ci/.github/workflows/ci.yml (hunk 1 above)
Comment (JonRowe) [reply ci/.github/workflows/ci.yml:2]: Yeah agreed.

Comment (benoittgt) [reply ci/.github/workflows/ci.yml:2]: Something like? I am not sur for pull_request.

```yml
on:
  # Trigger the workflow on push or pull request,
  # but only for the main branch
  push:
    branches:
      - main
      - *-maintenance
      - *-dev
  pull_request:
      branches:
        - **
```

File: ci/.github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          - 2.7
+          - 2.6
+          - 2.5
+          - 2.4
+          - 2.3
+          - 2.2
+          - 2.1.9
+          - ruby-head
+      fail-fast: false
+    continue-on-error: ${{ matrix.ruby == 'jruby-9.2.13.0' || endsWith(matrix.ruby, 'head') }}
```
Comment (benoittgt) [inline ci/.github/workflows/ci.yml:20]: Is it still needed?

Comment (JonRowe) [reply ci/.github/workflows/ci.yml:20]: head at least is needed post 3.0.0's release, and the jruby one is needed until its fixed

**Comment by pirj** [comment]
@JonRowe @benoittgt We have this left:
```
    - rvm: 2.7.1
      env: DIFF_LCS_VERSION="~> 1.3.0"
    - rvm: 2.7.1
      env: DIFF_LCS_VERSION="1.4.3"
```
Are we moving those to GHA? I've missed your `diff-lcs` adventure completely.

**Comment by JonRowe** [comment]
@pirj yes when I figure it out :joy:

**Comment by JonRowe** [comment]
Windows is coming in hot though 🔥

**Comment by pirj** [comment]
🕺

**Pull request merged**